phylogenetic/rules/prepare_sequences.smk

```diff
@@ -30,22 +30,28 @@ rule filter:
       - excluding strains in {input.exclude}
     """
     input:
-        sequences = "../../ingest/results/sequences.fasta",
-        metadata = "../../ingest/results/metadata.tsv",
-        exclude = lambda w: config["build_params"][w.build]["filter"]["exclude"],
-        include = lambda w: config["build_params"][w.build]["filter"]["include"],
+        sequences = lambda w: path_or_url(config["inputs"][0]['sequences']),
+        metadata = lambda w: path_or_url(config["inputs"][0]['metadata']),
+        # include/exclude are optional and the input values are handled as params
+        # (the `config_path` helper doesn't allow the config values to be optional)
+        exclude = lambda w: config["build_params"][w.build]["filter"].get("exclude", []),
+        include = lambda w: config["build_params"][w.build]["filter"].get("include", []),
     output:
         sequences = "results/{build}/filtered.fasta",
         metadata = "results/{build}/filtered.tsv",
         log = "results/{build}/filter-log.txt",
     params:
         id_column = config["id_column"],
-        min_length = lambda w: conditional("--min-length", config["build_params"][w.build]["filter"].get("min_length")),
-        min_date = lambda w: conditional("--min-date", config["build_params"][w.build]["filter"].get("min_date")),
-        max_date = lambda w: conditional("--max-date", config["build_params"][w.build]["filter"].get("max_date")),
-        exclude_where = lambda w: conditional("--exclude-where", config["build_params"][w.build]["filter"].get("exclude_where")),
-        group_by = lambda w: conditional("--group-by", config["build_params"][w.build]["filter"].get("group_by")),
-        subsample_max_sequences = lambda w: conditional("--subsample-max-sequences", config["build_params"][w.build]["filter"].get("subsample_max_sequences")),
+        min_length = conditional_config("--min-length", "filter", "min_length"),
+        min_date = conditional_config("--min-date", "filter", "min_date"),
+        max_date = conditional_config("--max-date", "filter", "max_date"),
```

Comment:
Similarly to https://github.com/nextstrain/ebola/pull/19#discussion_r2395951275, I think splitting the definition of config keys between a function and its parameters is confusing. I think the previous (now renamed) function is sufficient:

```py
min_length = lambda w: conditional_arg("--min-length", config["build_params"][w.build]["filter"].get("min_length")),
min_date = lambda w: conditional_arg("--min-date", config["build_params"][w.build]["filter"].get("min_date")),
max_date = lambda w: conditional_arg("--min-date", config["build_params"][w.build]["filter"].get("max_date")),
…
```

Reply:
(see discussion in other thread https://github.com/nextstrain/ebola/pull/19#discussion_r2395951275)